pkg/manager/member/pd_member_manager.go

```diff
@@ -97,6 +98,8 @@ func (m *pdMemberManager) syncPDServiceForTidbCluster(tc *v1alpha1.TidbCluster)
 	ns := tc.GetNamespace()
 	tcName := tc.GetName()
 
+	metrics.ClusterSpecReplicas.WithLabelValues(ns, tcName, "pd").Set(float64(tc.Spec.PD.Replicas))
```

Comment:
Please move this to L77.

Reply:
Do you need to collect the replicas of Pump and TiCDC?

Reply:
Is it OK that we collect the metrics here https://github.com/pingcap/tidb-operator/blob/master/pkg/controller/tidbcluster/tidb_cluster_control.go#L139 instead of in each member manager function? If yes, we also have to check nil of each component, e.g. `if spec.pd !=nil`.

Reply:
> Please use tc.PDStsDesiredReplicas() which will include the failover Pods.

No, this metric is similar to [kube_deployment_spec_replicas](https://github.com/kubernetes/kube-state-metrics/blob/master/internal/store/deployment.go#L170). We only care about the `spec.Replicas`.